src/components/character-count/character-count.js

```diff
@@ -0,0 +1,226 @@
+import '../../vendor/polyfills/Function/prototype/bind'
+import '../../vendor/polyfills/Event' // addEventListener and event.target normaliziation
+
+function CharacterCount ($module) {
+  this.$module = $module
+  this.$textarea = $module.querySelector('.js-character-count')
+}
+
+CharacterCount.prototype.defaults = {
+  characterCountAttribute: 'data-maxlength',
+  wordCountAttribute: 'data-maxwords'
+}
+
+// Initialize component
+CharacterCount.prototype.init = function (options) {
+  // Check for module
+  var $module = this.$module
+  var $textarea = this.$textarea
+  if (!$textarea) {
+    return
+  }
+
+  // Read options set using dataset ('data-' values) and merge them with options
+  var dataset = this.getDataset($module)
+  this.options = this.merge(options || {}, dataset)
+
+  // Determine the limit attribute (characters or words)
+  var countAttribute = (this.options.maxwords) ? this.defaults.wordCountAttribute : this.defaults.characterCountAttribute
```

Comment:
Hmmm we also allow setting `data-maxwords` in `template.njk`. What do you think about passing through all the `options` as data attributes from the markup instead?

Reply:
You can pass any attribute/option via data attributes or options - a merge happens on line 25.
Updated the PR description to reflect that.

Reply:
Thanks Alex. Is there an advantage to allowing user to do one or the other? I'm thinking that in terms of writing guidance it might be clearer if we just stick to one approach.

Reply:
I would recommend the data attribute as it has more visibility for developers. Init could be missed or placed in a more obscure place inside the project.

Reply:
That sounds sensible 👍 Are you happy to amend the component so that options can no longer be passed through from `init`?

Reply:
My personal choice would be to allow both and recommend one of them. Sometimes a dev may not have full access to how the markup is being rendered, in which case passing the attributes via javascript might come in handy.

Reply:
Updated: not the options can only be passed via data attributes.